### ISSUE TITLE
Fix doxygen warnings

### ISSUE THREAD

**Issue by t-b** [tango-controls/cppTango#532]
```
warning: doxygen no longer ships with the FreeSans font.
You may want to clear or change DOT_FONTNAME.
Otherwise you run the risk that the wrong font is being used for dot generated graphs.
warning: tag INPUT: input source `/home/firma/devel/cppTango/cppapi/server/tango_const.h' does not exist
warning: source /home/firma/devel/cppTango/cppapi/server/tango_const.h is not a readable file or directory... skipping.
/home/firma/devel/cppTango/cppapi/client/group.h:849: warning: found */ without matching /* while inside a \code block! Perhaps a missing \endcode?

Searching for include files...
Searching for example files...
Searching for images...
Searching for dot files...
Searching for msc files...
Searching for dia files...
Searching for files to exclude
Searching INPUT for files to process...
Searching for files in directory /home/firma/devel/cppTango/log4tango
Reading and parsing tag files
Parsing files
Preprocessing /home/firma/devel/cppTango/cppapi/server/device.h...
Parsing file /home/firma/devel/cppTango/cppapi/server/device.h...
Preprocessing /home/firma/devel/cppTango/cppapi/server/deviceclass.h...
Parsing file /home/firma/devel/cppTango/cppapi/server/deviceclass.h...
Preprocessing /home/firma/devel/cppTango/cppapi/server/command.h...
Parsing file /home/firma/devel/cppTango/cppapi/server/command.h...
Preprocessing /home/firma/devel/cppTango/cppapi/server/utils.h...
Parsing file /home/firma/devel/cppTango/cppapi/server/utils.h...
Preprocessing /home/firma/devel/cppTango/cppapi/server/except.h...
Parsing file /home/firma/devel/cppTango/cppapi/server/except.h...
Preprocessing /home/firma/devel/cppTango/cppapi/server/attrdesc.h...
Parsing file /home/firma/devel/cppTango/cppapi/server/attrdesc.h...
Preprocessing /home/firma/devel/cppTango/cppapi/server/fwdattrdesc.h...
Parsing file /home/firma/devel/cppTango/cppapi/server/fwdattrdesc.h...
Preprocessing /home/firma/devel/cppTango/cppapi/server/pipedesc.h...
Parsing file /home/firma/devel/cppTango/cppapi/server/pipedesc.h...
Preprocessing /home/firma/devel/cppTango/cppapi/server/attribute.h...
Parsing file /home/firma/devel/cppTango/cppapi/server/attribute.h...
Preprocessing /home/firma/devel/cppTango/cppapi/server/multiattribute.h...
Parsing file /home/firma/devel/cppTango/cppapi/server/multiattribute.h...
Preprocessing /home/firma/devel/cppTango/cppapi/server/pollext.h...
Parsing file /home/firma/devel/cppTango/cppapi/server/pollext.h...
Preprocessing /home/firma/devel/cppTango/cppapi/server/device_2.h...
Parsing file /home/firma/devel/cppTango/cppapi/server/device_2.h...
Preprocessing /home/firma/devel/cppTango/cppapi/server/w_attribute.h...
Parsing file /home/firma/devel/cppTango/cppapi/server/w_attribute.h...
Preprocessing /home/firma/devel/cppTango/cppapi/server/pipe.h...
Parsing file /home/firma/devel/cppTango/cppapi/server/pipe.h...
Preprocessing /home/firma/devel/cppTango/cppapi/server/w_pipe.h...
Parsing file /home/firma/devel/cppTango/cppapi/server/w_pipe.h...
Preprocessing /home/firma/devel/cppTango/cppapi/server/device_3.h...
Parsing file /home/firma/devel/cppTango/cppapi/server/device_3.h...
Preprocessing /home/firma/devel/cppTango/cppapi/server/device_4.h...
Parsing file /home/firma/devel/cppTango/cppapi/server/device_4.h...
Preprocessing /home/firma/devel/cppTango/cppapi/server/device_5.h...
Parsing file /home/firma/devel/cppTango/cppapi/server/device_5.h...
Preprocessing /home/firma/devel/cppTango/cppapi/server/encoded_attribute.h...
Parsing file /home/firma/devel/cppTango/cppapi/server/encoded_attribute.h...
Preprocessing /home/firma/devel/cppTango/cppapi/server/attrprop.h...
Parsing file /home/firma/devel/cppTango/cppapi/server/attrprop.h...
Preprocessing /home/firma/devel/cppTango/cppapi/client/dbapi.h...
Parsing file /home/firma/devel/cppTango/cppapi/client/dbapi.h...
Preprocessing /home/firma/devel/cppTango/cppapi/client/apiexcept.h...
Parsing file /home/firma/devel/cppTango/cppapi/client/apiexcept.h...
Preprocessing /home/firma/devel/cppTango/cppapi/client/devapi.h...
Parsing file /home/firma/devel/cppTango/cppapi/client/devapi.h...
Preprocessing /home/firma/devel/cppTango/cppapi/client/doc.h...
Parsing file /home/firma/devel/cppTango/cppapi/client/doc.h...
Preprocessing /home/firma/devel/cppTango/cppapi/client/devasyn.h...
Parsing file /home/firma/devel/cppTango/cppapi/client/devasyn.h...
Preprocessing /home/firma/devel/cppTango/cppapi/client/event.h...
Parsing file /home/firma/devel/cppTango/cppapi/client/event.h...
Preprocessing /home/firma/devel/cppTango/cppapi/client/group.h...
Parsing file /home/firma/devel/cppTango/cppapi/client/group.h...
Preprocessing /home/firma/devel/cppTango/cppapi/server/ta/home/firma/devel/cppTango/cppapi/server/attrdesc.h:413: warning: Member Attr(const char *name, Tango::DispLevel disp=Tango::OPERATOR) (function) of class Tango::Attr is not documented.
/home/firma/devel/cppTango/cppapi/server/attribute.h:1786: warning: Member set_min_alarm(char *new_min_alarm) (function) of class Tango::Attribute is not documented.
/home/firma/devel/cppTango/cppapi/server/attribute.h:1787: warning: Member set_min_alarm(const char *new_min_alarm) (function) of class Tango::Attribute is not documented.
/home/firma/devel/cppTango/cppapi/server/attribute.h:1812: warning: Member set_max_alarm(char *new_max_alarm) (function) of class Tango::Attribute is not documented.
/home/firma/devel/cppTango/cppapi/server/attribute.h:1813: warning: Member set_max_alarm(const char *new_max_alarm) (function) of class Tango::Attribute is not documented.
/home/firma/devel/cppTango/cppapi/server/attribute.h:1838: warning: Member set_min_warning(char *new_min_warning) (function) of class Tango::Attribute is not documented.
/home/firma/devel/cppTango/cppapi/server/attribute.h:1839: warning: Member set_min_warning(const char *new_min_warning) (function) of class Tango::Attribute is not documented.
/home/firma/devel/cppTango/cppapi/server/attribute.h:1864: warning: Member set_max_warning(char *new_max_warning) (function) of class Tango::Attribute is not documented.
/home/firma/devel/cppTango/cppapi/server/attribute.h:1865: warning: Member set_max_warning(const char *new_max_warning) (function) of class Tango::Attribute is not documented.
ngo.h...
Parsing file /home/firma/devel/cppTango/cppapi/server/tango.h...
Preprocessing /home/firma/devel/cppTango/cppapi/server/seqvec.h...
Parsing file /home/firma/devel/cppTango/cppapi/server/seqvec.h...
Building group list...
Building directory list...
Building namespace list...
Building file list...
Building class list...
Associating documentation with classes...
Computing nesting relations for classes...
Building example list...
Searching for enumerations...
Searching for documented typedefs...
Searching for members imported via using declarations...
Searching for included using directives...
Searching for documented variables...
Building interface member list...
Building member list...
Searching for friends...
Searching for documented defines...
Computing class inheritance relations...
Computing class usage relations...
Flushing cached template relations that have become invalid...
Computing class relations...
Add enum values to enums...
Searching for member function documentation...
Creating members for template instances...
Building page list...
Search for main page...
Computing page relations...
Determining the scope of groups...
Sorting lists...
Freeing entry tree
Determining which enums are documented
Computing member relations...
Building full member lists recursively...
Adding members to member groups.
Computing member references...
Inheriting documentation...
Generating disk names...
Adding source references...
Adding xrefitems...
Sorting member lists...
Computing dependencies between directories...
Generating citations page...
Counting data structures...
Resolving user defined references...
Finding anchors and sections in the documentation...
Transferring function references...
Combining using relations...
Adding members to index pages...
Generating style sheet...
Generating search indices...
Generating example documentation...
Generating file sources...
Parsing code for file apiexcept.h...
Parsing code for file attrdesc.h...
Parsing code for file attribute.h...
Parsing code for file attrprop.h...
Parsing code for file command.h...
Parsing code for file dbapi.h...
Parsing code for file devapi.h...
Parsing code for file devasyn.h...
Parsing code for file device.h...
Parsing code for file device_2.h...
Parsing code for file device_3.h...
Parsing code for file device_4.h...
Parsing code for file device_5.h...
Parsing code for file deviceclass.h...
Parsing code for file doc.h...
Parsing code for file encoded_attribute.h...
Parsing code for file event.h...
Parsing code for file except.h...
Parsing code for file fwdattrdesc.h...
Parsing code for file group.h...
Parsing code for file multiattribute.h...
Parsing code for file pipe.h...
Parsing code for file pipedesc.h...
Parsing code for file pollext.h...
Parsing code for file seqvec.h...
Parsing code for file tango.h...
Parsing code for file utils.h...
Parsing code for file w_attribute.h...
Parsing code for file w_pipe.h...
Generating file documentation...
Generating page documentation...
Generating docs for page except...
Generating docs for page recon...
Generating docs for page deprecated...
Generating group documentation...
Generating dependency graph for group Server
Generating dependency graph for group DBase
Generating dependency graph for group Client
Generating dependency graph for group Grp
Generating class documentation...
Generating namespace index...
Generating docs for namespace Tango
Generating docs for compound Tango::ApiUtil...
Generating docs for compound Tango::ArchiveEventInfo...
Generating docs for compound Tango::Attr...
Generating docs for compound Tango::AttrConfEventData...
Generating docs for compound Tango::AttrHistoryStack...
Generating docs for compound Tango::Attribute...
Generating docs for compound Tango::AttributeAlarmInfo...
Generating docs for compound Tango::AttributeEventInfo...
Generating docs for compound Tango::AttributeInfo...
Generating docs for compound Tango::AttributeInfoEx...
Generating docs for compound Tango::AttributeProxy...
Generating docs for compound Tango::AttrProp...
Generating docs for compound Tango::AttrR/home/firma/devel/cppTango/cppapi/server/command.h:1116: warning: Member insert(DevVarBooleanArray &data) (function) of class Tango::Command is not documented.
/home/firma/devel/cppTango/cppapi/client/Database.h:373: warning: Unsupported xml/html tag <instance_name> found
/home/firma/devel/cppTango/cppapi/server/attrdesc.h:790: warning: Member ImageAttr(const char *name) (function) of class Tango::ImageAttr is not documented.
/home/firma/devel/cppTango/cppapi/server/attrdesc.h:681: warning: Member SpectrumAttr(const char *_n) (function) of class Tango::SpectrumAttr is not documented.
/home/firma/devel/cppTango/cppapi/server/attrdesc.h:413: warning: Member Attr(const char *name, Tango::DispLevel disp=Tango::OPERATOR) (function) of class Tango::Attr is not documented.
/home/firma/devel/cppTango/cppapi/server/attrdesc.h:790: warning: Member ImageAttr(const char *name) (function) of class Tango::ImageAttr is not documented.
/home/firma/devel/cppTango/cppapi/server/attrdesc.h:681: warning: Member SpectrumAttr(const char *_n) (function) of class Tango::SpectrumAttr is not documented.
/home/firma/devel/cppTango/cppapi/server/attrdesc.h:413: warning: Member Attr(const char *name, Tango::DispLevel disp=Tango::OPERATOR) (function) of class Tango::Attr is not documented.
/home/firma/devel/cppTango/cppapi/server/attrdesc.h:681: warning: Member SpectrumAttr(const char *_n) (function) of class Tango::SpectrumAttr is not documented.
/home/firma/devel/cppTango/cppapi/server/attrdesc.h:413: warning: Member Attr(const char *name, Tango::DispLevel disp=Tango::OPERATOR) (function) of class Tango::Attr is not documented.
/home/firma/devel/cppTango/cppapi/server/command.h:1116: warning: Member insert(DevVarBooleanArray &data) (function) of class Tango::Command is not documented.
/home/firma/devel/cppTango/cppapi/server/command.h:1116: warning: Member insert(DevVarBooleanArray &data) (function) of class Tango::Command is not documented.
/home/firma/devel/cppTango/cppapi/server/command.h:2218: warning: Unsupported xml/html tag <typename> found
/home/firma/devel/cppTango/cppapi/server/command.h:1116: warning: Member insert(DevVarBooleanArray &data) (function) of class Tango::Command is not documented.
/home/firma/devel/cppTango/cppapi/server/command.h:1116: warning: Member insert(DevVarBooleanArray &data) (function) of class Tango::Command is not documented.
/home/firma/devel/cppTango/cppapi/server/command.h:2770: warning: Unsupported xml/html tag <typename> found
/home/firma/devel/cppTango/cppapi/server/w_attribute.h:117: warning: Member set_min_value(char *min_value) (function) of class Tango::WAttribute is not documented.
/home/firma/devel/cppTango/cppapi/server/w_attribute.h:118: warning: Member set_min_value(const char *min_value) (function) of class Tango::WAttribute is not documented.
/home/firma/devel/cppTango/cppapi/server/w_attribute.h:143: warning: Member set_max_value(char *max_value) (function) of class Tango::WAttribute is not documented.
/home/firma/devel/cppTango/cppapi/server/w_attribute.h:144: warning: Member set_max_value(const char *max_value) (function) of class Tango::WAttribute is not documented.
/home/firma/devel/cppTango/cppapi/server/attribute.h:1786: warning: Member set_min_alarm(char *new_min_alarm) (function) of class Tango::Attribute is not documented.
/home/firma/devel/cppTango/cppapi/server/attribute.h:1787: warning: Member set_min_alarm(const char *new_min_alarm) (function) of class Tango::Attribute is not documented.
/home/firma/devel/cppTango/cppapi/server/attribute.h:1812: warning: Member set_max_alarm(char *new_max_alarm) (function) of class Tango::Attribute is not documented.
/home/firma/devel/cppTango/cppapi/server/attribute.h:1813: warning: Member set_max_alarm(const char *new_max_alarm) (function) of class Tango::Attribute is not documented.
/home/firma/devel/cppTango/cppapi/server/attribute.h:1838: warning: Member set_min_warning(char *new_min_warning) (function) of class Tango::Attribute is not documented.
/home/firma/devel/cppTango/cppapi/server/attribute.h:1839: warning: Member set_min_warning(const char *new_min_warning) (function) of class Tango::Attribute is not documented.
/home/firma/devel/cppTango/cppapi/server/attribute.h:1864: warning: Member set_max_warning(char *new_max_warning) (function) of class Tango::Attribute is not documented.
/home/firma/devel/cppTango/cppapi/server/attribute.h:1865: warning: Member set_max_warning(const char *new_max_warning) (function) of class Tango::Attribute is not documented.
eadEvent...
Generating docs for compound Tango::AttrWrittenEvent...
Generating docs for compound Tango::AutoConnectTimeout...
Generating docs for compound Tango::CallBack...
Generating docs for compound Tango::ChangeEventInfo...
Generating docs for compound Tango::CmdDoneEvent...
Generating docs for compound Tango::CmdHistoryStack...
Generating docs for compound Tango::Command...
Generating docs for compound Tango::CommandInfo...
Generating docs for compound Tango::Connection...
Generating docs for compound Tango::Database...
Generating docs for compound Tango::DataElement...
Generating docs for compound Tango::DataReadyEventData...
Generating docs for compound Tango::DbClass...
Generating docs for compound Tango::DbDatum...
Generating docs for compound Tango::DbDevExportInfo...
Generating docs for compound Tango::DbDevFullInfo...
Generating docs for compound Tango::DbDevice...
Generating docs for compound Tango::DbDevImportInfo...
Generating docs for compound Tango::DbDevInfo...
Generating docs for compound Tango::DbHistory...
Generating docs for compound Tango::DbServer...
Generating docs for compound Tango::DbServerData...
Generating docs for compound Tango::DevCommandInfo...
Generating docs for compound Tango::Device_2Impl...
Generating docs for compound Tango::Device_3Impl...
Generating docs for compound Tango::Device_4Impl...
Generating docs for compound Tango::Device_5Impl...
Generating docs for compound Tango::DeviceAttribute...
Generating docs for compound Tango::DeviceAttributeConfig...
Generating docs for compound Tango::DeviceAttributeHistory...
Generating docs for compound Tango::DeviceClass...
Generating docs for compound Tango::DeviceData...
Generating docs for compound Tango::DeviceDataHistory...
Generating docs for compound Tango::DeviceImpl...
Generating docs for compound Tango::DevicePipe...
Generating docs for compound Tango::DevicePipeBlob...
Generating docs for compound Tango::DeviceProxy...
Generating docs for compound Tango::DevIntrChangeEventData...
Generating docs for compound Tango::DoubleAttrProp...
Generating docs for compound Tango::EncodedAttribute...
Generating docs for compound Tango::EventData...
Generating docs for compound Tango::Except...
Generating docs for compound Tango::FwdAttr...
Generating docs for compound Tango::Group...
Generating docs for compound Tango::GroupAttrReply...
Generating docs for compound Tango::GroupAttrReplyList...
Generating docs for compound Tango::GroupCmdReply...
Generating docs for compound Tango::GroupCmdReplyList...
Generating docs for compound Tango::GroupReply...
Generating docs for compound Tango::GroupReplyList...
Generating docs for compound Tango::ImageAttr...
Generating docs for compound Tango::MultiAttribute...
Generating docs for compound Tango::MultiAttrProp...
Generating docs for compound Tango::NamedDevFailed...
Generating docs for compound Tango::NamedDevFailedList...
Generating docs for compound Tango::PeriodicEventInfo...
Generating docs for compound Tango::Pipe...
Generating docs for compound Tango::PipeEventData...
Generating docs for compound Tango::PipeInfo...
Generating docs for compound Tango::SpectrumAttr...
Generating docs for compound Tango::TemplCommand...
Generating docs for compound Tango::TemplCommandIn...
Generating docs for compound Tango::TemplCommandInOut...
Generating docs for compound Tango::TemplCommandOut...
Generating docs for compound Tango::TimedAttrData...
Generating docs for compound Tango::TimedCmdData...
Generating docs for compound Tango::UserDefaultAttrProp...
Generating docs for compound Tango::UserDefaultFwdAttrProp...
Generating docs for compound Tango::UserDefaultPipeProp...
Generating docs for compound Tango::Util...
Generating docs for compound Tango::WAttribute...
Generating docs for compound Tango::WPipe...
Generating graph info page...
Generating directory documentation...
Generating index page...
Generating page index...
Generating module index...
Generating namespace member index...
Generating annotated compound index...
Generating alphabetical compound index...
Generating hierarchical class index...
Generating graphical class hierarchy...
Generating member index...
Generating file member index...
Generating example index...
finalizing index lists...
writing tag file...
Running plantuml with JAVA...
Running dot...
lookup cache used 4568/65536 hits=49300 misses=4807
finished...
Built target doc
```

**Comment by bourtemb** [comment]
Almost all the Doxygen warnings should be fixed in PR #544 
@t-b, could you please review this PR?
Only the following warning should remain:
`/home/firma/devel/cppTango/cppapi/client/group.h:849: warning: found */ without matching /* while inside a \code block! Perhaps a missing \endcode?`

It is due to the following code example in the Doxygen documentation:
```
 * @code
 * root_group->remove("*/stepper_motor/*");
 * @endcode
```

Doxygen is not clever enough on this one.

**Comment by bourtemb** [comment]
Hi @t-b,
Can we consider this issue solved?

As I wrote before:

> Only the following warning should remain:
> /home/firma/devel/cppTango/cppapi/client/group.h:849: warning: found */ without matching /* while inside a \code block! Perhaps a missing \endcode?
> 
> It is due to the following code example in the Doxygen documentation:
> 
>  * @code
>  * root_group->remove("*/stepper_motor/*");
>  * @endcode
> 
> Doxygen is not clever enough on this one.

I commented the doxygen ticket you mentioned (https://github.com/doxygen/doxygen/issues/6737) giving the above example.
I think the solution to remove this warning should come from the Doxygen team.
